src/thunderbots/software/ai/hl/stp/play/enemy_freekick_play.cpp

```diff
@@ -128,7 +128,10 @@ void EnemyFreekickPlay::getNextTactics(TacticCoroutine::push_type &yield)
         {
             shadow_tactic_main->updateParams(
                 enemy_threats.at(1), world.field(), world.friendlyTeam(),
-                world.enemyTeam(), ROBOT_MAX_RADIUS_METERS * 3, enemy_team_can_pass);
+                world.enemyTeam(), ROBOT_MAX_RADIUS_METERS * 3, enemy_team_can_pass,
+                world.ball(),
+                Util::DynamicParameters::DefenseShadowEnemyTactic::ball_steal_speed
+                    .value());
```

Comment:
I think it's better to explicitly force all parameters to be provided each time. I'm assuming we would remove duplication by updating all possible parameters before the if-statement, and then only updating what's necessary in either case. I'm hesitant to do this because I feel it opens up opportunities to mess up conditional statements and accidentally not update all the parameters each tick.

Reply:
Sorry, I'm confused? I'd just replace each of these calls with one to `updateShadowTactic`. For example here we would call `updateShadowTactic(shadow_tactic_main, enemy_threats.at(1))`?

Reply:
But this relies on the other parameters having already been updated in a previous call? I don't like update functions "relying" on previous update functions being called in order for the tactic to work properly. We can sync up in person about this.

Reply:
I looked at this again and don't think an `update` function makes sense. In the if-statement where you say the parameters are duplicated, are the first and only time the tactic is updated.

Unless you want to split this into an update at the beginning, and then a simplified update function within the if-blocks (which I don't think is better / worth the effort), then I don't think this needs changing.